src/serde.rs

```diff
@@ -24,15 +23,15 @@ pub trait ErrorType {
 #[non_exhaustive]
 pub enum ErrorKind {
     NotSupported,
-}    
-
+    SourceBufferTooSmall,
+}
 
 pub trait OutputSize {
     const OUTPUT_SIZE: usize;
 }
 
-pub trait Serde : ErrorType {
+pub trait Serde: ErrorType {
```

Comment:
In this context `Serde` refers to serialization and deserialization generally, and not to the well-known crates.io `serde` crate.  Maybe we should choose a different name.